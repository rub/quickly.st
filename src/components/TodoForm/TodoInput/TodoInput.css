.TodoInput-input {
  background: none;
  border: none;
  border-bottom: 1px solid #A4ADB6;
  color: #FFF;
  font-family: Roboto;
  font-size: 1rem;
  padding: 0.382rem 1.618rem 0.382rem 0;
  width: 100%;
}

.TodoInput-input::placeholder {
  color: #A4ADB6;
}

.TodoInput-input:focus {
  outline: none;
}
