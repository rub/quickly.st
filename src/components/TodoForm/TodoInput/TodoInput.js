import React from 'react';
import PropTypes from 'prop-types';

import './TodoInput.css';


/**
 * The input field where user can write todos
 */
const TodoInput = (props) => (
  <div className="TodoInput">
    <input
      className="TodoInput-input"
      type="text"
      placeholder="Add new item"
      autoFocus
      value={props.value}
      onChange={props.onChange}
    />
  </div>
)

TodoInput.propTypes = {
  /** TodoInput must have a value */
  value: PropTypes.string.isRequired,

  /** TodoInput must have a function which stores the user input */
  onChange: PropTypes.func.isRequired,
}

export default TodoInput;
