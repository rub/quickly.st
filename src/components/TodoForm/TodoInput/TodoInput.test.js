import React from 'react';
import TodoInput from './TodoInput';
import renderer from 'react-test-renderer';


it('renders the input field without crashing', () => {
  const tree = renderer.create(
    <TodoInput value="I'm a todo" onChange={jest.fn()}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
