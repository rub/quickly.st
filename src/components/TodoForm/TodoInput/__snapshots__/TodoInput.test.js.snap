// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the input field without crashing 1`] = `
<div
  className="TodoInput"
>
  <input
    autoFocus={true}
    className="TodoInput-input"
    onChange={[Function]}
    placeholder="Add new item"
    type="text"
    value="I'm a todo"
  />
</div>
`;
